src/MonitoringFunctions/Functions/DownloadSh.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+
+namespace MonitoringFunctions.Functions
+{
+    public static class DownloadSh
```

Comment:
minor: this is essentially the same logic as the previous class, the only variant is the URI

Reply:
What do you suggest we do here?

Reply:
consider consolidating the logic in a single place, then leveraging it with the parameter sets for ps1 & sh, respectively.  Moving forward, enhancements & bug fixes happen in a single place

Reply:
I think that is what I did when I created a helper method `CheckAndReportUrlAccessAsync` and called it from both functions. That is where we will do the fixes and improvements, and both functions will be benefiting. Functions are called by azure, so I don't have the option to pass additional parameters.

Reply:
Sounds like you've already thought about this & have implemented the best possible solution.  I am happy.  Let's close on this one.